### ISSUE TITLE
[ENHANCEMENT] Internal link naming

### ISSUE THREAD

**Issue by onemenzel** [Zettlr/Zettlr#992]
## Description

Right now, if one wants to add an internal link in text while still having a coherent text, a user has to add a link and write the links text before or after it, leaving no trace of where in the text the link belongs.

Wiki syntaxes solve this by allow for renaming internal links. So when an article get's rendered, another text than the page's name is being shown. This could be used to improve Zettlr's export of iternal links while not breaking the text if a link is set within it.

## Proposed Changes

Allow for internal links to be renamed like this: `[[file-name|displayed text]]`. The displayed text would then be used when exporting to another format.

## Caveats

Maybe a new setting to respect this kind of linking and to preserve those links would be necessary.

## Additional Information

Prior art e/g at the [mediawiki syntax documentation](https://www.mediawiki.org/wiki/Help:Links#Internal_links).

**Comment by nathanlesage** [comment]
We have had this discussion already a few weeks ago. I am principally against this, out of the following reasons:

* In MediaWiki, `[[link|displayed text]]` is being used as a tool to convert them into valid HTML links between documents. In Markdown, that's what the `[displayed text](link)` syntax is for. Adding the ability to change the display will lead to confusion, because then in effect you'd have two different ways of writing standard HTML links, but they fulfill different functionality
* In Zettlr, internal links are actually a form of search, because other than "normal" markdown link, the link's contents are being used to fill the search area. In its current form it is super easy to use and super fuzzy, and my aim is to make this search functionality even better in the future. Adding the ability to add titles will, from a UX perspective, nudge people to see these links as way more rigid than they actually are, and in their current form, being this "raw", enables more creative use.
* These links shouldn't be exported in the way of exporting to a different document, they are only for internal use, hence it's not necessary to add this display text thing. PLUS it would open the gates to someone pushing me to implement a rendering for that, which would again push Zettlr further towards WYSIWYG, which I am strictly against.

So, TL;DR: Won't implement this, BUT if enough people do make compelling use-cases, I may reconsider in the future. But in this case, I'd like to warmly invite you over to the forum --> https://forum.zettlr.com
